### ISSUE TITLE
Incompatible chain-code error when using "yo hyperledger-composer:angular"

### ISSUE THREAD

**Issue by sanjay-saxena** [hyperledger/composer#959]
I am trying to generate Angular2 app using `yo hyperledger-composer:angular` command. I have already started `Hyperledger Fabric 1.0` and `digitalproperty-network` is already deployed to the instance. However, the command to generate Angular2 app gets stuck. When I abort(Ctrl-C) the command, I see the following error message:

```
^CError: Deployed chain-code (0.7.1) is incompatible with client (0.7.2-20170510135245)
    at queryChainCode.then (/usr/local/lib/node_modules/generator-hyperledger-composer/node_modules/composer-connector-hlfv1/lib/hlfconnection.js:498:27)
```

## Context
The following shows the version `Hyperledger Composer` version numbers:

```
$ composer -version

composer-cli                   v0.7.1
composer-admin                 v0.7.1
composer-client                v0.7.1
composer-common                v0.7.1
composer-runtime-hlf           v0.7.1
```
However, when I install `generator-hyperleger-composer`, it brings down the following:

```
/usr/local/lib
└─┬ generator-hyperledger-composer@0.7.2-20170510135245 
  ├─┬ composer-client@0.7.2-20170510135245 
  │ ├── composer-common@0.7.2-20170510135245  deduped
  │ ├─┬ composer-connector-hlf@0.7.2-20170510135245 
  │ │ ├── composer-common@0.7.2-20170510135245  deduped
  │ │ ├── composer-runtime-hlf@0.7.2-20170510135245 
```
Since we now have different versions -- 0.7.1 and  0.7.2-20170510135245  -- of libraries in play, I guess it is causing the aforementioned error.

## Expected Behavior
Angular2 app should be generated.

## Actual Behavior
See above

## Possible Fix
Consistent packaging of libraries across the entire toolset, maybe. Or, chain-code backward compatibility across patches -- <major.minor.patch-build#>.

## Your Environment

* Version used: 0.7.1
* Environment name and version (e.g. Chrome 39, node.js 5.4): Node v0.6.10
* Operating System and version (desktop or mobile): macOS 10.12.1
* Link to your project:

**Comment by sanjay-saxena** [comment]
Note that uninstalling and re-installing `composer-cli` using the following commands:

```
$ npm uninstall -g composer-cli
$ npm install -g composer-cli
```
did not help. I see the same issue when trying to generate the Angular2 app.

Also, note that I have been able to submit transactions successfully and observe changes to the world state. It's just the Angular2 app generation that is causing this issue.

**Comment by sanjay-saxena** [comment]
With 0.7.2, this issue has been resolved. However, it would be good to make sure that all the toolsets are depending on the same versions of components to avoid this issue. Or, have a good backward compatibility story.

**Comment by LNga** [comment]
@sanjay-saxena Could you give more information about how you solved the issue with 0.7.2?
